experimental/LieAlgebras/src/LieAlgebraModuleHom.jl

```diff
@@ -0,0 +1,289 @@
+#
+###############################################################################
+
+function Base.show(io::IO, ::MIME"text/plain", h::LieAlgebraModuleHom)
+  io = pretty(io)
+  println(io, LowercaseOff(), "Lie algebra module morphism")
+  print(io, Indent())
+  println(io, "from ", Lowercase(), domain(h))
+  print(io, "to   ", Lowercase(), codomain(h))
+  print(io, Dedent())
+end
+
+function Base.show(io::IO, h::LieAlgebraModuleHom)
+  io = pretty(io)
+  if get(io, :supercompact, false)
+    print(io, LowercaseOff(), "Lie algebra module morphism")
+  else
+    print(io, LowercaseOff(), "Lie algebra module morphism: ")
+    print(io, Lowercase(), domain(h), " -> ", Lowercase(), codomain(h))
+  end
+end
+
+###############################################################################
+#
+#   Comparison
+#
+###############################################################################
+
+function Base.:(==)(h1::LieAlgebraModuleHom, h2::LieAlgebraModuleHom)
+  return domain(h1) === domain(h2) &&
+         codomain(h1) === codomain(h2) &&
+         matrix(h1) == matrix(h2)
+end
+
+function Base.hash(f::LieAlgebraModuleHom, h::UInt)
+  b = 0x7b887214521c25b7 % UInt
+  h = hash(objectid(domain(f)), h)
+  h = hash(objectid(codomain(f)), h)
+  h = hash(matrix(f), h)
+  return xor(h, b)
+end
+
+###############################################################################
+#
+#   Image and kernel
+#
+###############################################################################
+
+@doc raw"""
+    image(h::LieAlgebraModuleHom, v::LieAlgebraModuleElem) -> LieAlgebraModuleElem
+
+Return the image of `v` under `h`.
+"""
+function image(
+  h::LieAlgebraModuleHom{T1,T2}, v::LieAlgebraModuleElem
+) where {T1<:LieAlgebraModule,T2<:LieAlgebraModule}
+  @req parent(v) === domain(h) "Domain mismatch"
+  return codomain(h)(_matrix(v) * h.matrix)
+end
+
+# TODO: image and kernel, once submodules are implemented
+
+###############################################################################
+#
+#   Map operations
+#
+###############################################################################
+
+@doc raw"""
+    compose(f::LieAlgebraModuleHom, g::LieAlgebraModuleHom) -> LieAlgebraModuleHom
+
+Return the composition of `f` and `g`, i.e. the homomorphism `h` such that
+`h(x) = g(f(x))` for all `x` in the domain of `f`.
+The codomain of `f` must be identical to the domain of `g`.
+"""
+function compose(
+  f::LieAlgebraModuleHom{T1,T2}, g::LieAlgebraModuleHom{T2,T3}
+) where {T1<:LieAlgebraModule,T2<:LieAlgebraModule,T3<:LieAlgebraModule}
+  @req codomain(f) === domain(g) "Composition: Maps are not compatible"
+  h = LieAlgebraModuleHom(domain(f), codomain(g), f.matrix * g.matrix; check=false)
+  if isdefined(f, :inverse_isomorphism) && isdefined(g, :inverse_isomorphism)
+    h.inverse_isomorphism = LieAlgebraModuleHom(
+      codomain(g),
+      domain(f),
+      g.inverse_isomorphism.matrix * f.inverse_isomorphism.matrix;
+      check=false,
+    )
```

Comment:
Should this also be here?
```suggestion
    )
    h.inverse_isomorphism.inverse_isomorphism = h
```